### PR TITLE
Various fix/update to modulemgr.h

### PR DESCRIPTION
**Pull request by yne**
Fix invalid module_stop definition
Add missing module_exit
Change module_* to function pointer

### CODE REVIEW

File: include/psp2/kernel/modulemgr.h [hunk 1]
```diff
@@ -37,23 +37,34 @@ typedef struct SceKernelSegmentInfo
 	SceUInt res;    //!< unused?
 } SceKernelSegmentInfo;
 
+typedef struct {
```
Comment (xerpi) [inline include/psp2/kernel/modulemgr.h]: I think the convention is to put the struct name after "typedef struct" too.

File: include/psp2/kernel/modulemgr.h [hunk 2]
```diff
@@ -37,23 +37,34 @@ typedef struct SceKernelSegmentInfo
...
+	SceUInt32 ptr;  //!< prel31 offset to the start of the function
+	SceUInt32 insn; //!< start of the function (80 B0 B0 B0), 1=cannot be unwound
+} SceKernelModuleExidx;
+
+typedef struct {
```
Comment (xerpi) [inline include/psp2/kernel/modulemgr.h]: Same here, struct name missing.

File: include/psp2/kernel/modulemgr.h [hunk 3]
```diff
@@ -37,23 +37,34 @@ typedef struct SceKernelSegmentInfo
...
+	SceUInt32 insn0;  //!< 
+	SceUInt32 insn1;  //!< 
+	SceUInt32 flags;  //!< often 0
+} SceKernelModuleExtab;
+
 typedef struct
 {
```
Comment (xerpi) [inline include/psp2/kernel/modulemgr.h]: Mind placing the `{` bracket on the same line as typedef struct and also adding the name?

File: include/psp2/kernel/modulemgr.h (hunk 1 above)
Comment (yne) [reply include/psp2/kernel/modulemgr.h]: `SceKernelLMOption`, `SceKernelULMOption` and `SceKernelModuleInfo` does not follow theses rules neither.
Should I also update them ?

File: include/psp2/kernel/modulemgr.h (hunk 3 above)
Comment (yne) [reply include/psp2/kernel/modulemgr.h]: `SceKernelSegmentInfo` and `SceKernelModuleInfo` does not follow these rules.
Should I also update them ?

**Comment by Rinnegatamante** [comment]
Where's the missing module_exit added?

**Comment by yne** [comment]
in the staging area, on my laptop, from two years ago.
By the time I get home and `grep -r` my (probably removed) vitasdk folder, you would had time to create a PR ;)

**Pull request merged**